core/rfb.js

```diff
@@ -1346,6 +1344,8 @@
                 if (!ret) { return ret; }  // need more data
             }
 
+            this._display.flip();
```

Comment:
It was easier this way. Otherwise we need triple buffering to avoid tearing. Something to experiment with in the future perhaps.